server/handler.go

```diff
@@ -339,6 +328,9 @@ func (h *Handler) doQuery(
 		}
 	}()
 
+	eg, egCtx := errgroup.WithContext(ctx)
```

Comment:
Don't think this change should affect performance much, but will benchmark it in the merge to dolt.